src/app/static/src/app/components/plots/choropleth/Choropleth.vue

```diff
@@ -0,0 +1,379 @@
+<template>
+    <div class="row">
+        <div :class="hideControls ? 'col-sm-6 col-md-8' : 'col-md-3'">
+            <slot></slot>
```

Comment:
This bit of annoyance is here so we can keep the file uploads in the SAP view within the footprint of the choro (so the Filters can tuck underneath them). Hence the fiddly css and needing to 'hide controls' before we're ready to display the map and filters.

Reply:
Hm, I've just realised that this isn't a great solution as keeping the tabs in a row about the choropleth means that they stay there with the filters and map wrap on smaller screens, rather than jumping down to be above the map... Not really sure what to do about that!

Reply:
Maybe we should keep the option to keep Choro and Choro-filters as two separate controls, and lose the <slot> in the choro. Choro could be configured to either include its own choro-filters or not (it could in case of Output). They'd just need to both be wired up to the same plottingSelections state. Actually, if we did that, it would have the advantage that we could only show both controls once a file was loaded, so we could avoid a lot of that horrible intitialising/hideControls logic. What do you think? Put that in here or in a separate ticket?

Reply:
I agree this is a bit weird; filters as a separate component sounds good, it will also have the benefit of breaking up this megalithic component. Happy for that to be on a fresh ticket though.